lib/kafka/transaction_manager.rb

```diff
@@ -248,14 +248,18 @@ def send_offsets_to_txn(offsets:, group_id:)
       end
 
       retry_error_handler do
-        send_response = transaction_coordinator.txn_offset_commit(
+        send_response = group_coordinator(group_id: group_id).txn_offset_commit(
           transactional_id: @transactional_id,
           group_id: group_id,
           producer_id: @producer_id,
           producer_epoch: @producer_epoch,
           offsets: offsets
         )
-        Protocol.handle_error(send_response.error_code)
+        send_response.errors.each do |tp|
```

Comment:
Whats happens if there are no errors available (which is not an empty array, aka: `nil`)  for `send_response`? I'm not sure what data structure this is to be fair.. ? But chaking what happens if it is null or nil?

Reply:
You're asking about internal Kafka protocol structure which is binary over TCP. I don't think it's possible Kafka sends invalid response. Basically we copy the style of this gem here and don't do additional checks.
If you want to know more about protocol, visit: https://kafka.apache.org/protocol